metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py

```diff
@@ -284,16 +300,30 @@ def get_mces_from_node(
                 parentOwnership=node_owners,
                 defaults=defaults,
                 ingestion_config=ingestion_config,
+                ctx=ctx,
             )
 
 
+def get_domain_class(
+    graph: Optional[DataHubGraph], domains: List[str]
+) -> models.DomainsClass:
+    domain_registry: DomainRegistry = DomainRegistry(
```

Comment:
domain registry should be an instance variable

Reply:
Initially, I made it the instance variable, however, it doesn't resolve the domain name to the urn in _domain_registry.get_domain_urn(domain)_ function call. It resolves the domain name to the urn in the constructor call.

To make it an instance variable either I need to change the behavior of DomainRegistry or I need to traverse the business glossary in advance to collect all domains and pass it to the cache argument at the time of construction.

I think changing the behavior of DomainRegistry would be good. Let me know your thoughts both of the solutions will take some time extra to refactor the existing code

Reply:
In that case, let's just leave it as-is - i've added a todo to the code